src/lib/generateDefinition.ts

```
@@ -113,25 +199,25 @@ async function extenderOnOffLight(endpoint: Zh.Endpoint): Promise<GeneratedExten
             }
         }
 
-        const argsStr = JSON.stringify(args);
         if (endpoint.getDevice().manufacturerID === zh.Zcl.ManufacturerCode.Philips) {
-            return [{extend: philipsLight(args), source: `philipsLight(${argsStr})`, lib: 'philips'}];
+            generated.push({extendFn: philipsLight, args, source: `philipsLight`, lib: 'philips'});
         } else {
-            return [{extend: m.light(args), source: `light(${argsStr})`}];
+            generated.push({extendFn: m.light, args, source: `light`});
         }
-    } else {
-        return [{extend: m.onOff({powerOnBehavior: false}), source: 'onOff({powerOnBehavior: false})'}];
     }
+
+    return generated;
 }
 
-async function extenderElectricityMeter(endpoint: Zh.Endpoint): Promise<GeneratedExtend[]> {
+async function extenderElectricityMeter(endpoints: Zh.Endpoint[]): Promise<GeneratedExtend[]> {
+    // TODO: Support multiple endpoints
+    const endpoint = endpoints[0];
```

Please throw an error if multiple endpoints are passed

---

Updated, please check.